django/contrib/admin/helpers.py

```diff
@@ -417,17 +417,43 @@ def fields(self):
                     "help_text": form_field.help_text,
                 }
 
+    def get_inline_add_text(self):
+        """
+        Return 'Add XXX' or 'Add another XXX' depending on related object
+        presence.
+        """
+        verbose_name = capfirst(self.opts.verbose_name)
+        parent_instance = self.formset.instance
+        fk = getattr(self.formset, "fk", None)
+
+        if parent_instance.pk is None or isinstance(fk, OneToOneField):
+            return gettext("Add %(verbose_name)s") % {"verbose_name": verbose_name}
```

Comment:
If we think "Add ..." works well then we should use this always
We shouldn't be adding a database query to achieve this feature